Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -953,6 +955,25 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         completionHandler?(nil)
     }
 
+    public func handleShutDown(completionHandler: ((Data?) -> Void)? = nil) {
+        Task {
+            let managers = try await NETunnelProviderManager.loadAllFromPreferences()
+
+            guard let manager = managers.first else {
+                return
+            }
+
+            manager.isOnDemandEnabled = false
+            try await manager.saveToPreferences()
+            try await manager.loadFromPreferences()
+
+            let error = NSError(domain: "com.duckduckgo.vpn", code: 0)
+            stopTunnel(with: error)
```

Comment:
The method name clashes with the non-recommended `stopTunnel(with:)` so we should probably go with a different name.